### PR TITLE
disabled range-loop-analysis for clang version 12.0.0 up to 12.1.0

### PR DESCRIPTION
**Pull request by hkernbach**
# With the introduction of clang version 12.0.0 (clang-1200.0.32.2) the warning "range-loop-analysis" is now
    # enabled by default or changed analysis behaviour. The compiler now complains about "wrong" usage of
    # VPackObjectIterator and VPackArrayIterator - which is not the case.
    #
    # See example here: https://github.com/arangodb/arangodb/pull/12730
    #
    # Since we're stopping compilation on Apple in any warning case, we need to temporary disable that warning and
    # check upcoming versions if the compiler behaviour will be fixed.

### CODE REVIEW

**Comment by hkernbach** [comment]
http://jenkins01:8080/view/PR/job/arangodb-matrix-pr/12020/

File: CMakeLists.txt [hunk 1]
```diff
@@ -917,6 +917,18 @@ elseif (CMAKE_COMPILER_IS_CLANG)
 
   set(BASE_FLAGS "-Wall -Wextra -Wno-unused-parameter -Wno-deprecated-declarations ${BASE_FLAGS}")
 
+  if(APPLE AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL "12.0" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS "12.1")
+    # With the introduction of clang version 12.0.0 (clang-1200.0.32.2) the warning "range-loop-analysis" is now
```
Comment (goedderz) [inline CMakeLists.txt:921]:
```suggestion
    # With the introduction of apple clang version 12.0.0 (clang-1200.0.32.2) the warning "range-loop-analysis" is now
```

**Review by goedderz — APPROVED** [review]
LGTM

File: CMakeLists.txt [hunk 2]
```diff
@@ -917,6 +917,18 @@ elseif (CMAKE_COMPILER_IS_CLANG)
 
   set(BASE_FLAGS "-Wall -Wextra -Wno-unused-parameter -Wno-deprecated-declarations ${BASE_FLAGS}")
 
+  if(APPLE AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL "12.0" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS "12.1")
```
Comment (dothebart) [inline CMakeLists.txt:920]: please use `if(DARWIN`

Comment (hkernbach) [inline CMakeLists.txt:920]:
```suggestion
  if(DARWIN AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL "12.0" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS "12.1")
```

Comment (hkernbach) [reply CMakeLists.txt]: done

**Review by dothebart — APPROVED** [review]

**Comment by hkernbach** [comment]
Jenkins green except one unrelated issue. So can be merged?

**Comment by hkernbach** [comment]
Additionally @dothebart  @KVS85  
Do we want to have backports for this?

**Pull request merged**